presto-phoenix/pom.xml

```diff
@@ -0,0 +1,174 @@
+<?xml version="1.0"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.prestosql</groupId>
+        <artifactId>presto-root</artifactId>
+        <version>305-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>presto-phoenix</artifactId>
+    <description>Presto - Phoenix Connector</description>
+    <packaging>presto-plugin</packaging>
+
+    <properties>
+        <air.main.basedir>${project.parent.basedir}</air.main.basedir>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.phoenix</groupId>
+            <artifactId>presto-phoenix-client-shaded</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.prestosql</groupId>
+            <artifactId>presto-base-jdbc</artifactId>
```

Comment:
I am not sure this relation of this connector with `presto-base-jdbc`. Can you instead copy code you need and adjust it for phoenix needs instead of using `presto-base-jdbc` partially. Then maybe you wouldn't need to map `PhoenixRecordReader` to `ResultsSet` but you could use `PhoenixRecordReader` directly to produce data.

I am afraid that such usage of `presto-base-jdbc` would make `presto-base-jdbc` to be harder to maintain and evolve.

What do you think?

Reply:
In your first review, you wrote 
>Also, why not to use base-jdbc connector to create phoenix-connector? We could extend base-> jdbc connector if needed.

:).  The fact is, Phoenix is a SQL layer on top of a NoSQL store, so it is a weird hybrid, so I can see arguments for either way.
Phoenix implements most of the jdbc interface, and supports reading through a jdbc `ResultSet`.  The code in base-jdbc does all the Type translation for us here - e.g. `StandardColumnMapping` and the `Read/WriteFunction` can all be reused here. 

Copying the code was the approach the first PR tried.  It was a lot uglier IMHO and wouldn't inherit improvements to base-jdbc.

Btw I removed usage of `PhoenixRecordReader` now, and create the `ResultSet` directly in PhoenixClient

Reply:
Hmm... My $0.02. What leads to more code maintenance in the future is an interesting question.

Copying the code down allows base-jdbc to be changed with more ease it would perhaps also allow for using a few more of the "HBase'y" features of Phoenix (that might not map cleanly to JDBC);
on the other hand we now have copies of the code lying around, which now also have to be maintained, and base-jdbc already does a lot of the work that this connector would have to duplicate otherwise.

Personally I'd err on the side of less code-duplication. We should look at Phoenix connector as a JDBC type connector. That way things are simplified and improvements to base-jdbc would be visible everywhere.

Reply:
In general, you are correct. My point was that if you are using API that is prepared for certain use case and your use case is very different then it may become difficult to maintain the API. Then changing API may benefit in one use case, but could be a problem in the other. However, it is always a trade-off and sometimes it is hard to tell what is better to reuse or to copy.

Here, I wanted to make sure we are not is such a place.

Reply:
>  I removed usage of PhoenixRecordReader now, and create the ResultSet directly in PhoenixClient

Nice